src/fu-engine.c

```diff
@@ -2792,7 +2792,16 @@ fu_engine_md_refresh_device_verfmt (FuEngine *self, FuDevice *device, XbNode *co
 	/* found and different to existing */
 	if (verfmt != FWUPD_VERSION_FORMAT_UNKNOWN &&
 	    fu_device_get_version_format (device) != verfmt) {
+		FwupdRelease *rel = fu_device_get_release_default (device);
 		fu_device_set_version_format (device, verfmt);
+		if (rel != NULL) {
+			const gchar *version_old = fwupd_release_get_version (rel);
+			guint64 tmp = fu_common_strtoull (version_old);
+			if (tmp > 0) {
+				g_autofree gchar *version_new = fu_common_version_from_uint32 (tmp, verfmt);
```

Comment:
not convinced here. Shouldn't we use the same logic we use for the device?

Reply:
Well releases don't have a `VersionFormat` attribute or a raw version right now.  So If you think we should go that far down, it's a little more plumbing.

Reply:
I do wonder if that's too far down the rabbithole. Can't we just use the same verfmt as the device in all cases?

Reply:
I did an experiment with setting it immediately when the releases were loaded in `fu_engine_set_release_from_appstream` but realized that this will possibly lead to inconsistent results for devices that don't actually have the `md-set-*` flags set.  So my thought is it's better to refresh the release versions when metadata changes instead.

If you think this is still not the right way, would you propose another approach?